services/management/ia/deps.go

```diff
@@ -0,0 +1,28 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package ia
+
+import "github.com/percona/pmm-managed/models"
+
+// aletringService is a subset of methods of ia.Service used by this package.
+// We use it instead of real type for testing and to avoid dependency cycle.
+type aletringService interface {
```

Comment:
shouldn't this be renamed to something like `channelsService`? what I understand from this PR is that we will have one `Service` with methods for all the IA APIs in `/ia/ia.go`which will satisfy different interfaces defined here for channels, rules, templates and alerts.

so calling the behaviour for channels api as alertingService is confusing

Reply:
> different interfaces defined here for channels, rules, templates and alerts.

I'm not sure that it makes sense. `aletringService is a subset of methods of ia.Service used by this package.`, so it operates package level dependencies, not service (aka stuct) level.

In general, I don't know what is better, because I'm not sure that we will have one `Service` for IA stuff. Let's wait for the next ticket where I will add a second API. Probably these things will be refactored.

Reply:
Not sure about all that yet, but at very least there is a typo :)

Reply:
Typo fixed